From 7f07b26dd896d47339832dfa000dad5db174cacf Mon Sep 17 00:00:00 2001
From: paranoidbashthot <paranoidnela@gmail.com>
Date: Sun, 27 Dec 2020 14:27:34 +0100
Subject: [PATCH] 	modified:   include/pci_ids/radeonsi_pci_ids.h 
 modified:   src/amd/addrlib/src/amdgpu_asic_addr.h 	modified:  
 src/amd/common/ac_gpu_info.c 	modified:   src/amd/common/amd_family.h 
 modified:   src/amd/llvm/ac_llvm_util.c 	modified:  
 src/amd/vulkan/radv_pipeline.c 	modified:  
 src/amd/vulkan/winsys/null/radv_null_winsys.c 	modified:  
 src/gallium/drivers/radeonsi/si_state_draw.cpp 	modified:  
 src/gallium/targets/d3dadapter9/description.c 	modified:  
 src/gallium/winsys/radeon/drm/radeon_drm_winsys.c

	modified:   src/amd/vulkan/radv_pipeline.c
---
 include/pci_ids/radeonsi_pci_ids.h               |  5 +++++
 src/amd/addrlib/src/amdgpu_asic_addr.h           |  6 +++++-
 src/amd/common/ac_gpu_info.c                     | 16 +++++++++++++++-
 src/amd/common/amd_family.h                      |  2 ++
 src/amd/llvm/ac_llvm_util.c                      |  2 ++
 src/amd/vulkan/radv_pipeline.c                   |  3 ++-
 src/amd/vulkan/winsys/null/radv_null_winsys.c    |  4 ++++
 src/gallium/drivers/radeonsi/si_state_draw.cpp   |  3 ++-
 src/gallium/targets/d3dadapter9/description.c    |  2 ++
 .../winsys/radeon/drm/radeon_drm_winsys.c        |  6 ++++++
 10 files changed, 45 insertions(+), 4 deletions(-)

diff --git a/include/pci_ids/radeonsi_pci_ids.h b/include/pci_ids/radeonsi_pci_ids.h
index 8c4567e8483..9f0d596f4ee 100644
--- a/include/pci_ids/radeonsi_pci_ids.h
+++ b/include/pci_ids/radeonsi_pci_ids.h
@@ -145,6 +145,11 @@ CHIPSET(0x131B, KAVERI)
 CHIPSET(0x131C, KAVERI)
 CHIPSET(0x131D, KAVERI)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x67A0, HAWAII)
 CHIPSET(0x67A1, HAWAII)
 CHIPSET(0x67A2, HAWAII)
diff --git a/src/amd/addrlib/src/amdgpu_asic_addr.h b/src/amd/addrlib/src/amdgpu_asic_addr.h
index 02b241e4fdb..07d1c33614d 100644
--- a/src/amd/addrlib/src/amdgpu_asic_addr.h
+++ b/src/amd/addrlib/src/amdgpu_asic_addr.h
@@ -71,7 +71,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF
 
@@ -122,6 +124,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)           ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)           ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 1ea04abd8b0..2fb88d96eac 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -261,6 +261,8 @@ static uint32_t get_l2_cache_size(enum radeon_family family)
       return 1024 * 1024;
    case CHIP_FIJI:
    case CHIP_POLARIS10:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
       return 2048 * 1024;
       break;
    default:
@@ -535,6 +537,8 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
       identify_chip2(SPOOKY, KAVERI);
       identify_chip2(KALINDI, KABINI);
       identify_chip2(GODAVARI, KABINI);
+      identify_chip2(STARSHA, LIVERPOOL);
+      identify_chip2(STARSHP, GLADIUS);
       break;
    case FAMILY_VI:
       identify_chip(ICELAND);
@@ -670,7 +674,7 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    info->pa_sc_tile_steering_override = device_info.pa_sc_tile_steering_override;
    info->max_render_backends = amdinfo->rb_pipes;
    /* The value returned by the kernel driver was wrong. */
-   if (info->family == CHIP_KAVERI)
+   if (info->family == CHIP_KAVERI || info->family == CHIP_LIVERPOOL)
       info->max_render_backends = 2;
 
    /* Guess the number of enabled SEs because the kernel doesn't tell us. */
@@ -1199,6 +1203,8 @@ int ac_get_gs_table_depth(enum chip_class chip_class, enum radeon_family family)
    case CHIP_PITCAIRN:
    case CHIP_VERDE:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_HAWAII:
    case CHIP_TONGA:
    case CHIP_FIJI:
@@ -1268,6 +1274,14 @@ void ac_get_raster_config(struct radeon_info *info, uint32_t *raster_config_p,
       raster_config = 0x3a00161a;
       raster_config_1 = 0x0000002e;
       break;
+   case CHIP_LIVERPOOL:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x00000000;
+      break;
+   case CHIP_GLADIUS:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x0000002e;
+      break;
    default:
       fprintf(stderr, "ac: Unknown GPU, using 0 for raster_config\n");
       raster_config = 0x00000000;
diff --git a/src/amd/common/amd_family.h b/src/amd/common/amd_family.h
index a6175e8b4fd..c4037b71204 100644
--- a/src/amd/common/amd_family.h
+++ b/src/amd/common/amd_family.h
@@ -82,6 +82,8 @@ enum radeon_family
    CHIP_HAINAN,
    CHIP_BONAIRE, /* GFX7 (Sea Islands) */
    CHIP_KAVERI,
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_KABINI,
    CHIP_HAWAII,
    CHIP_TONGA, /* GFX8 (Volcanic Islands & Polaris) */
diff --git a/src/amd/llvm/ac_llvm_util.c b/src/amd/llvm/ac_llvm_util.c
index 503b19c245e..baa1b54cc4b 100644
--- a/src/amd/llvm/ac_llvm_util.c
+++ b/src/amd/llvm/ac_llvm_util.c
@@ -132,6 +132,8 @@ const char *ac_get_llvm_processor_name(enum radeon_family family)
    case CHIP_HAINAN:
       return "hainan";
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
       return "bonaire";
    case CHIP_KABINI:
       return "kabini";
diff --git a/src/amd/vulkan/radv_pipeline.c b/src/amd/vulkan/radv_pipeline.c
index d48c2782d57..54554d3a77c 100644
--- a/src/amd/vulkan/radv_pipeline.c
+++ b/src/amd/vulkan/radv_pipeline.c
@@ -1477,7 +1477,8 @@ radv_compute_ia_multi_vgt_param_helpers(struct radv_pipeline *pipeline)
 		/* Bug with tessellation and GS on Bonaire and older 2 SE chips. */
 		if ((device->physical_device->rad_info.family == CHIP_TAHITI ||
 		     device->physical_device->rad_info.family == CHIP_PITCAIRN ||
-		     device->physical_device->rad_info.family == CHIP_BONAIRE) &&
+		     device->physical_device->rad_info.family == CHIP_BONAIRE ||
+		     device->physical_device->rad_info.family == CHIP_LIVERPOOL) /* ??*/ &&
 		    radv_pipeline_has_gs(pipeline))
 			ia_multi_vgt_param.partial_vs_wave = true;
 		/* Needed for 028B6C_DISTRIBUTION_MODE != 0 */
diff --git a/src/amd/vulkan/winsys/null/radv_null_winsys.c b/src/amd/vulkan/winsys/null/radv_null_winsys.c
index 22863f0ae7b..e26f440ed7f 100644
--- a/src/amd/vulkan/winsys/null/radv_null_winsys.c
+++ b/src/amd/vulkan/winsys/null/radv_null_winsys.c
@@ -43,6 +43,10 @@ static const struct {
 	[CHIP_OLAND] = { 0x6060, 2, true },
 	[CHIP_HAINAN] = { 0x6660, 2, true },
 	[CHIP_BONAIRE] = { 0x6640, 4, true },
+	[CHIP_LIVERPOOL] = { 0x9920, 8 },
+	[CHIP_LIVERPOOL] = { 0x9922, 8 },
+	[CHIP_LIVERPOOL] = { 0x9923, 8 },
+	[CHIP_GLADIUS] = { 0x9924, 16 },
 	[CHIP_KAVERI] = { 0x1304, 2, false },
 	[CHIP_KABINI] = { 0x9830, 2, false },
 	[CHIP_HAWAII] = { 0x67A0, 16, true },
diff --git a/src/gallium/drivers/radeonsi/si_state_draw.cpp b/src/gallium/drivers/radeonsi/si_state_draw.cpp
index 60acfda75ce..23a0e6bd7b7 100644
--- a/src/gallium/drivers/radeonsi/si_state_draw.cpp
+++ b/src/gallium/drivers/radeonsi/si_state_draw.cpp
@@ -366,7 +366,8 @@ static unsigned si_get_init_multi_vgt_param(struct si_screen *sscreen, union si_
 
       /* Bug with tessellation and GS on Bonaire and older 2 SE chips. */
       if ((sscreen->info.family == CHIP_TAHITI || sscreen->info.family == CHIP_PITCAIRN ||
-           sscreen->info.family == CHIP_BONAIRE) &&
+           sscreen->info.family == CHIP_BONAIRE ||
+	    sscreen->info.family == CHIP_LIVERPOOL) /* ? */ &&
           key->u.uses_gs)
          partial_vs_wave = true;
 
diff --git a/src/gallium/targets/d3dadapter9/description.c b/src/gallium/targets/d3dadapter9/description.c
index a3e4cd61770..09104f49101 100644
--- a/src/gallium/targets/d3dadapter9/description.c
+++ b/src/gallium/targets/d3dadapter9/description.c
@@ -53,6 +53,8 @@ cards_amd[] = {
     {"KAVERI",                      "AMD Radeon(TM) R7 Graphics"},
     {"KABINI",                      "AMD Radeon HD 8400 / R3 Series"},
     {"BONAIRE",                     "AMD Radeon HD 8770"},
+    {"LIVERPOOL",                   "AMD Liverpool (Playstation 4)"},
+    {"GLADIUS",                     "AMD Gladius (Playstation 4 Pro)"},
     {"OLAND",                       "AMD Radeon HD 8670"},
     {"HAINAN",                      "AMD Radeon HD 8600M Series"},
     {"TAHITI",                      "AMD Radeon HD 7900 Series"},
diff --git a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
index ef7434d7589..f9b1466239d 100644
--- a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
@@ -275,6 +275,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
       ws->info.chip_class = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -493,6 +495,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.num_tcc_blocks = 4;
       break;
@@ -522,9 +526,11 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
+      case CHIP_GLADIUS:
          ws->info.max_se = 4;
          break;
       }
-- 
2.25.1

